File: python/SampleFactory/inputclient.py
"""
inputclient.py
"""

from input import *
from inputfactory import *

def main():
    factory = InputFactory()

    input = factory.create("file")
    input.read()

    input = factory.create("console")
    input.read()

    input = factory.create("net")
    input.read()

if __name__ == '__main__':
    main()

File: cc/SampleFactory/inputclient.cc
#include "input.h"
#include "inputfactory.h"

int main(int argc, char* argv[])
{
    InputFactory factory;

    Input* input = factory.create(FILE_INPUT);
    input->read();
    delete input;

    input = factory.create(CONSOLE_INPUT);
    input->read();
    delete input;

    input = factory.create(NET_INPUT);
    input->read();
    delete input;

    return 0;
}

